service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/relayNamespace/CreateAzureRelayNamespaceStepTest.java

```diff
@@ -0,0 +1,177 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.relayNamespace;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.cloudres.azure.resourcemanager.relay.data.CreateRelayRequestData;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.common.BaseAzureTest;
+import bio.terra.workspace.common.fixtures.ControlledResourceFixtures;
+import bio.terra.workspace.generated.model.ApiAzureIpCreationParameters;
+import bio.terra.workspace.generated.model.ApiAzureRelayNamespaceCreationParameters;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.management.Region;
+import com.azure.core.management.exception.ManagementError;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.core.util.Context;
+import com.azure.resourcemanager.relay.RelayManager;
+import com.azure.resourcemanager.relay.models.Namespaces;
+import com.azure.resourcemanager.relay.models.RelayNamespace;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("azure")
+public class CreateAzureRelayNamespaceStepTest extends BaseAzureTest {
+
+  private static final String STUB_STRING_RETURN = "stubbed-return";
+
+  @Mock private FlightContext mockFlightContext;
+  @Mock private CrlService mockCrlService;
+  @Mock private AzureConfiguration mockAzureConfig;
+  @Mock private AzureCloudContext mockAzureCloudContext;
+  @Mock private RelayManager mockRelayManager;
+  @Mock private Namespaces mockNamespaces;
+  @Mock private RelayNamespace.DefinitionStages.Blank mockStage1;
+  @Mock private RelayNamespace.DefinitionStages.WithResourceGroup mockStage2;
+  @Mock private RelayNamespace.DefinitionStages.WithCreate mockStage3;
+  @Mock private RelayNamespace mockRelayNamespace;
+  @Mock private ManagementException mockException;
+  @Mock private FlightMap mockWorkingMap;
+
+  private ArgumentCaptor<Context> contextCaptor = ArgumentCaptor.forClass(Context.class);
+
+  @BeforeEach
+  public void setup() {
+    when(mockAzureCloudContext.getAzureResourceGroupId()).thenReturn(STUB_STRING_RETURN);
```

Comment:
Why does the cloud context need to be mocked given it's a POJO?

Reply:
I think the intent is just that this is not valid azure cloud context...admittedly, I've just been copy pasting from other tests...maybe `fakeAzureCloudContext` will be a bit better?